osu.Game/Beatmaps/IO/ArchiveReader.cs

```diff
@@ -12,27 +12,40 @@ namespace osu.Game.Beatmaps.IO
 {
     public abstract class ArchiveReader : IDisposable, IResourceStore<byte[]>
     {
+        protected enum ArchiveStorageMethod
+        {
+            Directory, File
+        }
+
         private class Reader
         {
             public Func<BasicStorage, string, bool> Test { get; set; }
             public Type Type { get; set; }
+            public ArchiveStorageMethod ArchiveStorageMethod { get; set; }
         }
 
         private static List<Reader> readers { get; } = new List<Reader>();
-    
+
         public static ArchiveReader GetReader(BasicStorage storage, string path)
         {
             foreach (var reader in readers)
             {
-                if (reader.Test(storage, path))
-                    return (ArchiveReader)Activator.CreateInstance(reader.Type, storage.GetStream(path));
+                if (!reader.Test(storage, path)) continue;
+
+                switch (reader.ArchiveStorageMethod)
+                {
+                    case ArchiveStorageMethod.Directory:
+                        return (ArchiveReader)Activator.CreateInstance(reader.Type, path);
+                    default:

```

Comment:
I would prefer `case ArchiveStorageMethod.File:` here. In my opinion, it will be easier to read and understand the code.